src/main/java/org/broadinstitute/hellbender/tools/spark/sv/GatherSplitReadsSpark.java

```diff
@@ -0,0 +1,664 @@
+                    overlappers.add(entry);
+            }
+            if ( overlappers.size() >= MIN_EVIDENCE )
+            {
+                final List<GATKRead> list = new LinkedList<>();
+                for ( Map.Entry<EventLocus,GATKRead> entry : overlappers )
+                {
+                    if (entry.getValue() != null)
+                    {
+                        list.add(entry.getValue());
+                        entry.setValue(null);
+                    }
+                }
+                return list.iterator();
+            }
+            return Collections.emptyIterator();
+        }
+    }
+
+    // Class that acts as a mapper from a stream of reads to a stream of <Kmer,BreakpointId> pairs.
+    // Sitting atop the BreakpointClusterer, it groups funky reads that are mapped near each other and assigns that
+    // group a breakpoint id, and transforms each funky read into kmers which are emitted along with the breakpoint id.
+    private static final class ClusterKmerizer implements Iterator<Tuple2<Kmer,Long>>, Iterable<Tuple2<Kmer,Long>>
+    {
+        private final Iterator<GATKRead> inputIterator;
+        private final Set<Kmer> killSet;
+        private Iterator<Tuple2<Kmer,Long>> outputIterator = Collections.emptyIterator();
+        private String currentContig = null;
+        private int lastLocus = 0;
+        private Long breakpointId;
+
+        private static final int CLUSTER_BREAK_DISTANCE = 500;
+        //TODO: Should depend on fragment size
+
+        public ClusterKmerizer( Integer partIdx, Iterator<GATKRead> someInputIterator, Set<Kmer> someKillSet )
+        {
+            inputIterator = someInputIterator;
+            killSet = someKillSet;
+            breakpointId = partIdx.longValue() << 32;
+        }
+
+        @Override
+        public boolean hasNext()
+        {
+            while ( !outputIterator.hasNext() )
+            {
+                if ( !inputIterator.hasNext() )
+                    return false;
+                outputIterator = processRead(inputIterator.next());
+            }
+            return true;
+        }
+
+        @Override
+        public Tuple2<Kmer,Long> next()
+        {
+            if ( !hasNext() )
+                throw new NoSuchElementException("ClusterKmerizer is exhausted.");
+            return outputIterator.next();
+        }
+
+        @Override
+        public Iterator<Tuple2<Kmer,Long>> iterator() { return this; }
+
+        private Iterator<Tuple2<Kmer,Long>> processRead( GATKRead read )
+        {
+            if ( read.getContig() != currentContig || read.getStart() > lastLocus+CLUSTER_BREAK_DISTANCE )
+                breakpointId += 1;
+            currentContig = read.getContig();
+            lastLocus = read.getStart();
+
+            List<Tuple2<Kmer,Long>> kmerList = new ArrayList<>(read.getLength()-KLEN+1);
+            BytesKmerizer kmerizer = new BytesKmerizer(read.getBases(),KLEN);
```

Comment:
once BytesKmerizer is Iterable this can be re-written more cleanly using a stream